elements/lisk-p2p/src/p2p.ts

```diff
@@ -406,6 +412,29 @@ export class P2P extends EventEmitter {
 						INVALID_CONNECTION_SELF_REASON,
 					);
 
+					// Ignore these steps for local network tests since all the IPs will be localhost
+					if (socket.remoteAddress !== '127.0.0.1') {
+						const selfWSPort = queryObject.wsPort
+							? +queryObject.wsPort
+							: this._nodeInfo.wsPort;
+						const selfP2PInfo = {
+							ipAddress: socket.remoteAddress,
+							wsPort: selfWSPort,
+						};
+						// When a node tries to connect itself then add it to blacklist so that it doesn't connect itself.
+						this._config.blacklistedPeers = this._config.blacklistedPeers
```

Comment:
Maybe `blacklistedPeers` should be stored as a private property on the `P2P` class (and always use that one instead of the one on the config)? That way we don't need to mutate the config object.